src/playground/sample_data_ingestion/sample_data_ingestor/sample_data_ingestor.py

```diff
@@ -0,0 +1,345 @@
+    feature, 
+    timestamp, 
+    number, 
+    avg_min, 
+    avg_max, 
+    abs_min, 
+    abs_max, 
+    anomaly_percentage = .001, 
+    mapping = True, 
+    minutes = 2,
+    current_date = datetime.today() - timedelta(days = 7)
+):
+    """
+    Function that generates data that simulates a trend with or without anomalies
+
+    Arguments:
+        - data_template: The template (JSON, mapping, CSV)
+        - feature: The field that will be used to test anomalies
+        - timestamp: The field name that contains timestamps
+        - number: How many documents to generate
+        - avg_min: The lowest average value in which there is no anomaly
+        - avg_max: The highest average value in which there is no anomaly
+        - abs_min: The lowest possible value, which can be an anomaly
+        - abs_max: The highest possible value, which can be an anomaly
+        - anomaly_percentage: How often an anomaly occurs (minimum: 0.001)
+        - mapping: Whether or not the input is a mapping
+        - minutes: The time interval for each data point (e.g. if minutes = 2, this tool will generate entries with timestamps that are 2 minutes apart from one another)
+        - current_date: The date at which entries are generated
+
+    Returns:
+        - An array of generated data, which simulates a data with a trend with some anomalies
+    """
+    dataset = []
+
+    for i in range(number):
+        entry = generate_data(data_template, mapping) 
+        milliseconds = int(current_date.strftime("%s")) * 1000
+        # If entry came from an input file, load data to the dataset list
+        if type(entry) is list:
+            for element in entry:
+                element = loads(element)
+                element[feature] = generate_noise(element[feature], anomaly_percentage, avg_min, avg_max, abs_min, abs_max)
+                element[timestamp] = milliseconds
+                dataset.append(element)
+        else:
+            entry = loads(entry)
+            entry[feature] = generate_noise(entry[feature], anomaly_percentage, avg_min, avg_max, abs_min, abs_max)
+        
+            # Update timestamp with current time
+            entry[timestamp] = milliseconds
+            dataset.append(dumps(entry))
+        # Increment current date
+        current_date += timedelta(minutes = minutes)
+    return dataset
+
+
+def ingest_from_user_data(filename):
+    """
+    Utility function that loads user provided data to a list
+
+    Arguments:
+        - filename: The name of the user provided data file as a .json or .csv file (zipped or unzipped)
+
+    Returns:
+        - A list of data loaded from the file
+
+    Raises:
+        TypeError: Filename is not a string or is not a valid filename
+    """
+
+    # Input validation
+    if type(filename) is not str or (type(filename) is str and "." not in filename):
+        raise TypeError("Filename is not a string or is not a valid filename")
+    name = filename.split(".gz")[0]
```

Comment:
if the filename is .tar or .txt, what's happening here?

Reply:
I added an extra line at the end, which raises a `ValueError` in case .json or .csv files were not provided